include/circt/Dialect/FIRRTL/FIRRTLUtils.h

```diff
@@ -146,17 +146,16 @@ inline FIRRTLBaseType getBaseType(Type type) {
 }
 
 /// Return base type or passthrough if FIRRTLType, else null.
-inline FIRRTLBaseType getBaseTypeOrNull(Type type) {
-  auto ftype = dyn_cast_or_null<FIRRTLType>(type);
-  if (!ftype)
+inline FIRRTLBaseType getBaseTypeIfPresent(Type type) {
+  if (!type)
     return {};
-  return getBaseType(ftype);
+  return getBaseType(type);
 }
 
 /// Get base type if isa<> the requested type, else null.
 template <typename T>
 inline T getBaseOfType(Type type) {
-  return dyn_cast_or_null<T>(getBaseTypeOrNull(type));
+  return dyn_cast_or_null<T>(getBaseTypeIfPresent(type));
```

Comment:
Hmm this function is only ever called on `port.getType()` and never passed null. 
```suggestion
  return dyn_cast_or_null<T>(getBaseType(type));
```